### PR TITLE
draft "security.txt" spec integration

### PR DESCRIPTION
**Pull request by austinheap**
#### TLDR;

This PR contains a basic implementation of the [`security.txt` spec](https://spec.securitytext.org/) ([IETF](https://tools.ietf.org/html/draft-foudil-securitytxt-04)). More than happy to correct any issues, just let me know.

#### Routes
* `/security.txt` => 301 redirect to `/.well-known/security.txt`
* `/.well-known/security.txt` => 200 `text/plain` with basic `security.txt` document

#### Config
* New key: `instance.securitytxt` -- used as template for `security.txt` document
* Old key: `admin.email` -- used as default value for second `Contact` directive

#### Output
Of redirect route via `curl -is http://localhost:9000/security.txt`:
```sh
HTTP/1.1 301 Moved Permanently
X-DNS-Prefetch-Control: off
X-Frame-Options: DENY
Strict-Transport-Security: max-age=15552000; includeSubDomains
X-Download-Options: noopen
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: Retry-After
Tk: N
Location: /.well-known/security.txt
Vary: Accept
Content-Type: text/plain; charset=utf-8
Content-Length: 59
Date: Wed, 05 Sep 2018 17:21:09 GMT
Connection: keep-alive

Moved Permanently. Redirecting to /.well-known/security.txt
```

Of document route via `curl -is http://localhost:9000/.well-known/security.txt`:
```sh
HTTP/1.1 200 OK
X-DNS-Prefetch-Control: off
X-Frame-Options: DENY
Strict-Transport-Security: max-age=15552000; includeSubDomains
X-Download-Options: noopen
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: Retry-After
Tk: N
Content-Type: text/plain; charset=utf-8
Content-Length: 145
ETag: W/"91-wRj34aDyBEPd8NDpBHz0wEtgjuc"
Date: Wed, 05 Sep 2018 17:22:50 GMT
Connection: keep-alive

# If you would like to report a security issue
# you may report it to:
Contact: https://github.com/Chocobozzz/PeerTube
Contact: admin@example.com
```

(Thanks for all your hard work btw!)

### CODE REVIEW

**Comment by rigelk** [comment]
LGTM. That is also the opportunity to define a security patching and disclosure policy like https://disclose.io/ or https://github.com/bugcrowd/disclosure-policy or https://oss-security.openwall.org/wiki/disclosure/researcher

**Comment by austinheap** [comment]
Thanks for the quick feedback and good points. I only included the bare minimum directive (`Contact: `) and didn't include any of the others. If needed they can be added later, but users can always customize the template in their [YAML](https://github.com/Chocobozzz/PeerTube/pull/1020/files#diff-8e71c54313a9b8a1d6c20ceae3927846) [configs](https://github.com/Chocobozzz/PeerTube/pull/1020/files#diff-fe7044f2ecd69c76ce484ad03fabc12f) to add other directives. My line of thought was 1) if people discover issues they should easily know where to report them and 2) the best place is probably on GitHub [the first `Contact: ` directive] but 3) it should include the instance admin's email too [the second `Contact: ` directive].

The spec most certainly is new but has had early adoption from some pretty big orgs (security.txt on [google.com](https://www.google.com/.well-known/security.txt), [dropbox.com](https://www.dropbox.com/.well-known/security.txt), [facebook.com](https://www.facebook.com/.well-known/security.txt), [etc](https://securitytext.org/)). I maintain [other](https://github.com/austinheap/php-security-txt) [security.txt](https://github.com/austinheap/laravel-security-txt)-[related](https://github.com/austinheap/wordpress-security-txt) [projects](https://github.com/securitytext/data) and am more than happy to keep this up-to-date!

**Comment by austinheap** [comment]
@artlog typos fixed 🎉

**Comment by Nutomic** [comment]
Does that expose the admin's email in plain text by default? Seems kind of problematic because it invites spam. Might be a good idea to disable this by default, or point to the contact form from #745 once it's implemented.

**Comment by austinheap** [comment]
@Nutomic It does and I know there are a lot of differing opinions around this. I realize I'm a sample set of one, but my experience has been this: I used the same spam filters before and after my emails appeared in a handful of `security.txt` documents.

Nothing changed in how I handled spam, didn't get better, didn't get worse, didn't change a thing.

**Comment by austinheap** [comment]
Admin e-mails are already public but I agree everyone has different spam setups/concerns.

```bash
> DOMAIN="framatube.org" && \
  curl -s "https://$DOMAIN/feeds/videos.atom?videoChannelId=1" | \
  grep email | cut -d'>' -f2 | cut -d'<' -f1

< tech@framasoft.org
```

**Comment by austinheap** [comment]
Will do, thank you @rigelk!
